src/OrchardCore.Modules/OrchardCore.OpenId/Startup.cs

```diff
@@ -102,6 +105,66 @@ public override void ConfigureServices(IServiceCollection services)
             // Disabling same-site is required for OpenID's module prompt=none support to work correctly.
             services.ConfigureApplicationCookie(options => options.Cookie.SameSite = SameSiteMode.None);
         }
+
+        public override void Configure(IApplicationBuilder app, IRouteBuilder routes, IServiceProvider serviceProvider)
+        {
+            var openIddictServerOptions = serviceProvider.GetRequiredService<IOptions<OpenIddictServerOptions>>().Value;
```

Comment:
This won't work because OpenIddict uses named options (like any other authentication handler). Instead, use `IOptionsMonitor`:

```csharp
var options = serviceProvider.GetRequiredService<IOptionsMonitor<OpenIddictServerOptions>>()
    .Get(OpenIddictServerDefaults.AuthenticationScheme);
```